mutable_set/mutable_set.mbt

```diff
@@ -0,0 +1,568 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// MutableSet is an open-addressing hashset based on robin hood hash.
+pub struct MutableSet[T] {
+  priv mut directory : Array[Option[Item[T]]]
+  priv mut real_size : Int
+  priv mut effective_size : Int
+}
+
+priv struct Item[T] {
+  mut key : T
+  mut probe_length : Int
+} derive(Debug, Show)
+
+/// Initialize MutableSet by specifying initial capacity.
+pub fn MutableSet::with_capacity[T](initial_capacity : Int) -> MutableSet[T] {
+  {
+    directory: if initial_capacity <= 0 || initial_capacity > 2147483647 {
```

Comment:
is the check `initial_capacity > 2147483647` really necessary?

Reply:
In MoonBit, literals that are too long cannot be compiled. In addition, when it is larger than this value, it will overflow to a value < 0, so this is unnecessary